### ISSUE TITLE
`unused_async_trait_impl` generates incorrect suggestion for statements containing `return`

### ISSUE THREAD

**Issue by Wassasin** [rust-lang/rust-clippy#17179]
### Summary

`unused_async_trait_impl` now only wraps the tail expression of a body with `core::futures::ready(...)`. However when a function also contains a `return e;` statement, `e` in that statement should also be wrapped with `core::futures::ready(...)`.

In the suggestion this lint should also change all return statements within the HIR tree to return expressions wrapped in `core::futures::ready(...)`.

### Reproducer

Code:
```rust
struct Test;

impl crate::HasAsyncMethod for Test {
    async fn do_something() -> u32 {
        if true {
            return 3;
        }

        5
    }
}
```

Current output:
```rust
struct Test;

impl crate::HasAsyncMethod for Test {
    fn do_something() -> impl Future<Output = u32> {
        if true {
            return 3;
        }

        std::future::ready(5)
    }
}
```

Desired output:
```rust
struct Test;

impl crate::HasAsyncMethod for Test {
    fn do_something() -> impl Future<Output = u32> {
        if true {
            return std::future::ready(3);
        }

        std::future::ready(5)
    }
}
```

### Version

```text
rustc 1.98.0-nightly (57d06900f 2026-05-27)
binary: rustc
commit-hash: 57d06900fd7d9ee06d3a7f323bb77f17ab3cfaf8
commit-date: 2026-05-27
host: x86_64-unknown-linux-gnu
release: 1.98.0-nightly
LLVM version: 22.1.6
```

### Additional Labels

@rustbot label+I-suggestion-causes-error

**Comment by Wassasin** [comment]
@ada4a might also be of interest to you to track.
